ingestion/src/metadata/sdk/data_quality/runner.py

```diff
@@ -0,0 +1,439 @@
+        """
+        runner = cls(table_fqn, client=client)
+        runner._initialize()
+        return runner
+
+    @classmethod
+    def for_service(cls, table_fqn: str) -> Self:
+        """Alias for for_table(). Initialize runner for a specific table FQN.
+
+        Args:
+            table_fqn: Fully qualified name of the table (e.g., "MySQL.default.db.table")
+
+        Returns:
+            TestRunner instance
+        """
+        return cls.for_table(table_fqn)
+
+    @classmethod
+    def from_yaml_file(
+        cls,
+        file_path: str,
+        use_connection_from_yaml: bool = False,
+        client: Optional[OMeta[Any, Any]] = None,
+    ) -> Self:
+        """Build TestRunner from a YAML workflow file."""
+        with open(file_path, "r", encoding="utf-8") as stream:
+            return cls.from_yaml_string(stream.read(), use_connection_from_yaml, client)
+
+    @classmethod
+    def from_yaml_string(
+        cls,
+        string: str,
+        use_connection_from_yaml: bool = False,
+        client: Optional[OMeta[Any, Any]] = None,
+    ) -> Self:
+        """Build TestRunner from a YAML workflow string."""
+        data = yaml.safe_load(string)
+
+        config = OpenMetadataWorkflowConfig(**data)
+        source = config.source
+
+        assert (
+            source.type == TestSuiteConfigType.TestSuite.value
+        ), f"Can't create test suite for source type: {source.type}"
+
+        source_config = source.sourceConfig.config
+        assert isinstance(
+            source_config, TestSuitePipeline
+        ), f"Can't create test suite for source config type: {type(source.sourceConfig.config)}"
+        assert (
+            source_config.entityFullyQualifiedName is not None
+        ), "TestSuitePipeline config must have entity fully qualified name"
+
+        if use_connection_from_yaml:
+            client = OMeta(config=config.workflowConfig.openMetadataServerConfig)
+
+        runner = cls.for_table(
+            source_config.entityFullyQualifiedName.root, client=client
+        )
+
+        processor: Optional[TestSuiteProcessorConfig] = None
+        if config.processor and config.processor.config:
+            processor = TestSuiteProcessorConfig(**config.processor.config.model_dump())
+
+        if processor is None:
+            return runner
+
+        if tests_definitions := processor.testCases:
+            runner.test_definitions = tests_definitions
+
+        return runner
+
+    def _initialize(self) -> None:
+        """Fetch table entity and service connection from OpenMetadata.
+
+        This method retrieves the table entity and associated database service connection
+        from OpenMetadata. It validates that the table exists and has a properly configured
+        service connection.
+
+        Raises:
+            ValueError: If table not found, service not found, or connection not configured
+        """
+        self.table_entity = self.tables.get_by_name(
```

Comment:
[nit] can we rename `self.tables` and the like to `self.table_client` (so to have clarity that this is a client and not holding actual table objects.